chia/full_node/full_node.py

```diff
@@ -2216,45 +2221,48 @@ async def respond_transaction(
         if not test and not (await self.synced()):
             return MempoolInclusionStatus.FAILED, Err.NO_TRANSACTIONS_WHILE_SYNCING
 
+        spend_bundle_id = bytes_to_SpendBundleID(spend_name)
         if self.mempool_manager.get_spendbundle(spend_name) is not None:
-            self.mempool_manager.remove_seen(spend_name)
+            self.mempool_manager.remove_seen(spend_bundle_id)
             return MempoolInclusionStatus.SUCCESS, None
-        if self.mempool_manager.seen(spend_name):
+        if self.mempool_manager.seen(spend_bundle_id):
             return MempoolInclusionStatus.FAILED, Err.ALREADY_INCLUDING_TRANSACTION
-        self.mempool_manager.add_and_maybe_pop_seen(spend_name)
-        self.log.debug(f"Processing transaction: {spend_name}")
+        self.mempool_manager.add_and_maybe_pop_seen(spend_bundle_id)
+        self.log.debug(f"Processing transaction: {spend_bundle_id}")
         # Ignore if syncing
         if self.sync_store.get_sync_mode():
             status = MempoolInclusionStatus.FAILED
             error: Optional[Err] = Err.NO_TRANSACTIONS_WHILE_SYNCING
-            self.mempool_manager.remove_seen(spend_name)
+            self.mempool_manager.remove_seen(spend_bundle_id)
         else:
             try:
-                cost_result = await self.mempool_manager.pre_validate_spendbundle(transaction, tx_bytes, spend_name)
+                cost_result = await self.mempool_manager.pre_validate_spendbundle(
+                    transaction, tx_bytes, spend_bundle_id
+                )
             except ValidationError as e:
-                self.mempool_manager.remove_seen(spend_name)
+                self.mempool_manager.remove_seen(spend_bundle_id)
                 return MempoolInclusionStatus.FAILED, e.code
             except Exception:
-                self.mempool_manager.remove_seen(spend_name)
+                self.mempool_manager.remove_seen(spend_bundle_id)
                 raise
             async with self._blockchain_lock_low_priority:
                 if self.mempool_manager.get_spendbundle(spend_name) is not None:
```

Comment:
it would seem reasonable for `get_spendbundle()` to take a `SpendBundleID` too

Reply:
I did start doing that, but `get_spendbundle()` is called about 236 times, so that one has to wait.